src/ideal/ideal.jl

```diff
@@ -663,4 +676,64 @@ function minimal_generating_set(I::sideal)
    end
    return gens(Ideal(R, Singular.libSingular.idMinBase(I.ptr, R.ptr)))
 end
- 
+
+###############################################################################
+#
+#   Independent sets
+#
+###############################################################################
+
+@doc Markdown.doc"""
+    independent_sets(I::sideal{S}) where S <: Union{spoly{T}, spoly{n_unknown{T}}} where T <: Union{Singular.FieldElem, Nemo.FieldElem}
+> Returns all non-extendable independent sets of $lead(I)$. $I$ has to be given
+> by a Gröbner basis.
+"""
+function independent_sets(I::sideal{S}) where S <: Union{spoly{T}, spoly{n_unknown{T}}} where T <: Union{Singular.FieldElem, Nemo.FieldElem}
+   I.isGB == false && error("I needs to be a Gröbner basis.")
+   R = base_ring(I)
+   n = nvars(R)
+   a = Array{Int32, 1}()
+   libSingular.scIndIndset(I.ptr, R.ptr, a, true)
+   m = Int(div(length(a), n))
+   a = Int.(transpose(reshape(a, n, m)))
+   P = Array{Array{spoly, 1}, 1}()
+   for i in 1:m
+      b  = Array{spoly, 1}()
+      for j in findall(x->x == 1, a[i, :])
+         push!(b, gen(R, j))
+      end
+      push!(P, b)
+   end
+   return P
+end
+
+@doc Markdown.doc"""
+    maximal_independent_set(I::sideal{S}; all::Bool = false) where S <: Union{spoly{T}, spoly{n_unknown{T}}} where T <: Union{Singular.FieldElem, Nemo.FieldElem}
+> Returns, by default, an array containing a maximal independet set of
+> $lead(I)$. $I$ has to be given by a Gröbner basis.
+> If the additional parameter "all" is set to true, an array containing
+> all maximal independent sets iof $lead(I)$ is returned.
+"""
+function maximal_independent_set(I::sideal{S}; all::Bool = false) where S <: Union{spoly{T}, spoly{n_unknown{T}}} where T <: Union{Singular.FieldElem, Nemo.FieldElem}
```

Comment:
I think you want S <: Union{spoly{T}, spoly{n_unknown{U}} where {T <: Singular.FieldElem, U <: Nemo.FieldElem}

Reply:
That's true. I will fix it and prepare the next pull request.